File: List.c
// Menu driven program to show operations in list :-

#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node* next;
};

struct Node* head = NULL;

void insertAtEnd(int value) {
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = value;
    newNode->next = NULL;
    
    if (head == NULL) {
        head = newNode;
        return;
    }
    
    struct Node* temp = head;
    while (temp->next != NULL) {
        temp = temp->next;
    }
    temp->next = newNode;
}

void deleteNode(int value) {
    struct Node* temp = head, *prev = NULL;
    
    if (temp != NULL && temp->data == value) {
        head = temp->next;
        free(temp);
        return;
    }
    
    while (temp != NULL && temp->data != value) {
        prev = temp;
        temp = temp->next;
    }
    
    if (temp == NULL) return;
    
    prev->next = temp->next;
    free(temp);
}

void displayList() {
    struct Node* temp = head;
    while (temp != NULL) {
        printf("%d -> ", temp->data);
        temp = temp->next;
    }
    printf("NULL\n");
}

int searchList(int value) {
    struct Node* temp = head;
    while (temp != NULL) {
        if (temp->data == value)
            return 1;
        temp = temp->next;
    }
    return 0;
}

int main() {
    int choice, value;
    do {
        printf("\nMenu:\n");
        printf("1. Insert\n");
        printf("2. Delete\n");
        printf("3. Display\n");
        printf("4. Search\n");
        printf("5. Exit\n");
        printf("Enter your choice: ");
        scanf("%d", &choice);

        switch (choice) {
            case 1:
                printf("Enter value to insert: ");
                scanf("%d", &value);
                insertAtEnd(value);
                break;
            case 2:
                printf("Enter value to delete: ");
                scanf("%d", &value);
                deleteNode(value);
                break;
            case 3:
                displayList();
                break;
            case 4:
                printf("Enter value to search: ");
                scanf("%d", &value);
                if (searchList(value))
                    printf("Value found in list.\n");
                else
                    printf("Value not found in list.\n");
                break;
            case 5:
                printf("Exiting...\n");
                break;
            default:
                printf("Invalid choice! Please try again.\n");
        }
    } while (choice != 5);
    
    return 0;
}

File: List.py
# Menu driven program to show operations in list :-

class Node:
    """A node in the linked list"""
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:
    """Linked List with basic operations"""
    def __init__(self):
        self.head = None

    def insert(self, data):
        """Insert a node at the end of the list"""
        new_node = Node(data)
        if not self.head:
            self.head = new_node
        else:
            temp = self.head
            while temp.next:
                temp = temp.next
            temp.next = new_node
        print(f"{data} inserted into the list.")

    def delete(self, key):
        """Delete a node by value"""
        temp = self.head

        if temp and temp.data == key:
            self.head = temp.next
            temp = None
            print(f"{key} deleted from the list.")
            return

        prev = None
        while temp and temp.data != key:
            prev = temp
            temp = temp.next

        if not temp:
            print(f"{key} not found in the list.")
            return

        prev.next = temp.next
        temp = None
        print(f"{key} deleted from the list.")

    def display(self):
        """Display the linked list"""
        temp = self.head
        if not temp:
            print("The list is empty.")
            return
        print("Linked List:", end=" ")
        while temp:
            print(temp.data, end=" -> ")
            temp = temp.next
        print("None")

    def search(self, key):
        """Search for an element in the list"""
        temp = self.head
        pos = 1
        while temp:
            if temp.data == key:
                print(f"{key} found at position {pos}.")
                return
            temp = temp.next
            pos += 1
        print(f"{key} not found in the list.")

def menu():
    """Display menu and handle user input"""
    linked_list = LinkedList()

    while True:
        print("\nMenu:")
        print("1. Insert")
        print("2. Delete")
        print("3. Display")
        print("4. Search")
        print("5. Exit")
        
        choice = input("Enter your choice: ")

        if choice == '1':
            data = int(input("Enter the value to insert: "))
            linked_list.insert(data)
        elif choice == '2':
            key = int(input("Enter the value to delete: "))
            linked_list.delete(key)
        elif choice == '3':
            linked_list.display()
        elif choice == '4':
            key = int(input("Enter the value to search: "))
            linked_list.search(key)
        elif choice == '5':
            print("Exiting the program.")
            break
        else:
            print("Invalid choice! Please enter a number between 1 and 5.")

if __name__ == "__main__":
    menu()
